### PR TITLE
C|NIM|CORE: Add request/reply ID

### PR DESCRIPTION
**Pull request by Jeff-Ciesielski**
This will allow responses to specific messages rather than message types.

Needs tested on the kiosk.  Surface level stuff seems to work, but you know how that goes...

@keyme/robotics-engineers

### CODE REVIEW

File: XBVC/emitters/templates/c_header.jinja2 [hunk 1]
```diff
@@ -52,6 +52,9 @@ void xbvc_init(void *params,
 	       xbvc_platform_init_handler platform_init);
 
 void xbvc_run(void);
-int xbvc_send(void *msg, uint32_t msg_type);
-	
+int xbvc_send(void *msg, uint32_t msg_type, uint32_t *rnd_id_ret);
```
Comment (knitHacker) [inline XBVC/emitters/templates/c_header.jinja2]: Is there a reason the xbvc_send and xbvc_send_response can't generate the random number?

File: XBVC/emitters/templates/c_interface.jinja2 [hunk 2]
```diff
@@ -332,6 +369,11 @@ int xbvc_send(void *msg, uint32_t msg_type)
     return ret;
 }
 
+int xbvc_send(void *msg, uint32_t msg_type, uint32_t *rnd_id_ret)
+{
+    return xbvc_send_response(msg, msg_type, 0, rnd_id_ret);
```
Comment (knitHacker) [inline XBVC/emitters/templates/c_interface.jinja2]: is 0 a special number that isn't part of the generated ids? Could a dispatcher see 0 in this field and know it's not a response?

**Comment by knitHacker** [comment]
Talked about my comments offline and 0 or another constant will be a special case so we can decide if it is a response or not and we pass in the random number in c so we have access to it in the handlers. 

LGTM

**Comment by Jeff-Ciesielski** [comment]
Updated commentary and re-tested

**Comment by Jeff-Ciesielski** [comment]
@keyme/robotics-engineers : heads up - update your XBVC after this.

**Pull request merged**